instrumentation/armeria-1.3/library/src/main/java/io/opentelemetry/instrumentation/armeria/v1_3/ArmeriaTracingBuilder.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.armeria.v1_3;
+
+import io.opentelemetry.api.OpenTelemetry;
+import java.util.Collections;
+import java.util.Map;
+
+/** A builder of {@link ArmeriaTracing}. */
+public final class ArmeriaTracingBuilder {
+  private final OpenTelemetry openTelemetry;
+  private Map<String, String> peerServiceMapping = Collections.emptyMap();
+
+  ArmeriaTracingBuilder(OpenTelemetry openTelemetry) {
+    this.openTelemetry = openTelemetry;
+  }
+
+  /**
+   * Used to specify a mapping from hostnames or IP addresses (key of the passed {@code
+   * peerServiceMapping} map) to peer services (value of the map). The peer service is added as an
+   * attribute to a span whose host or IP match the mapping.
+   *
+   * @see <a
+   *     href="https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/span-general.md#general-remote-service-attributes">peer.service
+   *     attribute specification</a>
+   */
+  public ArmeriaTracingBuilder setPeerServiceMapping(Map<String, String> peerServiceMapping) {
```

Comment:
plural seems slightly better to me, but maybe too late to change b/c of env var `OTEL_INSTRUMENTATION_COMMON_PEER_SERVICE_MAPPING`
```suggestion
  public ArmeriaTracingBuilder setPeerServiceMappings(Map<String, String> peerServiceMappings) {
```